packages/core/src/indexing/postgres/init-tables.ts

```diff
@@ -21,27 +25,36 @@ export async function listMidTables(dataSource: Knex): Promise<Array<string>> {
   return midTables
 }
 
+function relationsDefinitionsToColumnInfo(relations?: ModelRelationsDefinition): Array<ColumnInfo> {
+  if (!relations) {
+    return []
+  }
+  return Object.keys(relations).map((keyName) => {
+    return { name: keyName, type: ColumnType.STRING }
+  })
+}
+
 /**
  * Create mid tables and corresponding indexes
  */
 export async function initTables(
   dataSource: Knex,
-  modelsToIndex: Array<StreamID>,
+  modelsToIndex: Array<IndexModelArgs>,
   logger: DiagnosticsLogger
 ) {
-  const expectedTables = modelsToIndex.map(asTableName).map((tableName) => {
+  for (const modelIndexArgs of modelsToIndex) {
+    const tableName = asTableName(modelIndexArgs.model)
     if (tableName.length > 63) {
-      logger.err(`Invalid model added to config file: ${tableName}`)
-      process.exit(-1)
```

Comment:
I got rid of this `process.exit` call and changed it to just throw a normal Error.  I was afraid of this as a potential DOS security risk - if someone sends a request to the admin API to index a model with a long streamid now you can force the node to crash.  They'd need to have signed the request with an admin key, but still made me nervous. Just failing the current request instead of the whole process seems totally reasonable to me anyway

Reply:
Yeah, I'd agree that throwing the error is reasonable now especially in the context of the admin API - the exit was also tied to the config file update necessitating a node restart so that we catch any potential model misconfiguration actively.